daemon-tests/tests/liquidation.rs

```diff
@@ -80,11 +78,15 @@ async fn given_rollover_when_oracle_attests_long_liquidation_price_can_liquidate
     );
 
     let oracle_data_rollover = OliviaData::example_1();
-    mock_oracle_announcements(&mut maker, &mut taker, oracle_data_rollover.announcements()).await;
-
-    taker
-        .trigger_rollover_with_latest_dlc_params(order_id)
-        .await;
+    rollover(
+        &mut maker,
+        &mut taker,
+        order_id,
+        oracle_data_rollover.clone(),
+        None,
+        fee_calculator.complete_fee_for_rollover_hours(24),
+    )
+    .await;
```

Comment:
🙃 I'm not too fond of this because it makes the test a bit more complicated with all this data that needs to be passed around needlessly (we don't actually care about all the things that are being asserted in the `rollover` helper in this test).

I did depend on this helper in a previous version too, but I purposely moved away from it. I would prefer to call `wait_next_state` a couple of times explicitly here to save you all the other changes in this patch and still fix the bug in the test.

Reply:
Alternatively, we could split the `rollover` helper into two, so that one purely helps you to execute the protocol (like the `open_cfd` helper) and the other helps you assert things (not needed here).

Reply:
> we don't actually care about all the things that are being asserted in the rollover helper in this test

Is this worth worrying about? On the one hand, the tests are a bit less isolated (i.e if you break rollover, this is guaranteed to break too), but on the other hand it possibly extends the level of integration in the tests (if you break rollover subtly in a way that does not cause the test to fail for whatever reason, this test will still fail).

Reply:
> Is this worth worrying about?

I worry about it because all the needless complexity in the test makes it harder to understand and maintain for us in the future. As an example: why is this test talking about a `fee_calculator`, when none of that matters for what we're trying to verify here?

> On the one hand, the tests are a bit less isolated (i.e if you break rollover, this is guaranteed to break too)

I actually don't care so much about the fact that this test will fail when rollover is broken. In an ideal world we wouldn't have to deal with that minimal noise, but I can happily live with that.

> but on the other hand it possibly extends the level of integration in the tests (if you break rollover subtly in a way that does not cause the test to fail for whatever reason, this test will still fail)

This can be a valid argument depending on the case. We have some places where we check that rollover works after something new has happened (for example, after retrying a rollover so that the taker can catch up), and I think that's a good idea.

But here there is nothing to verify: we are running a regular contract setup and rollover as part of the _arrange_ phase of the test.

Reply:
> I'm not too fond of this because it makes the test a bit more complicated with all this data that needs to be passed around needlessly (we don't actually care about all the things that are being asserted in the rollover helper in this test).

I generally agree with this, and I am not a fan or introducing the `fee_calculator` into the liquidation tests. I will see to split the rollover function. 
Besides that I think it is great to have an abstraction for "do a rollover" in the tests, that can be used by tests other than the pure rollover tests. But I think you don't have a problem with that :)